### ISSUE TITLE
Extract no-else-return for except from `no-else-return` into a separate rule

### ISSUE THREAD

**Issue by marcinnformula** [pylint-dev/pylint#9008]
### Current problem

Due to #7788 solved via GH-7888 valid `try-except-else` raises warning from `no-else-return` when there is return in `except` block.

The code below is valid and readable by explicit definition of a `else` block:

```python
def test_exception(raise_exception=True):
    try:
        if raise_exception:
            raise ValueError
    except ValueError:
        return "exception raised"
    else:
        return "exception NOT raised"
```

After #7788 Pylint warns that the code should be changed to something like this:

```python
def test_exception(raise_exception=True):
    try:
        if raise_exception:
            raise ValueError
    except ValueError:
        return "exception raised"
    return "exception NOT raised"
```

The first form has two major advantages and it is used widely:
* gives better readability (PEP20) due to verbose `except` and `else` blocks
* avoids negative cons due to possible refactoring in except block when removing return (which would change the program flow).

It would be nice to have a separate rule for exceptions, which would allow users to disable it on demand.  I understand that the concept is similar to `if-else`, but `try-except` is more complex. When you're adding or removing `finally` block you must change `else` block all the time, which is error prone / unnecessary from a programming perspective. So in such cases it would be nice to have `no-else-return` enabled for `if-else`, and (something like) `no-except-else-return` disabled for `try-except` blocks.

CC: @Pierre-Sassoulas 

### Desired solution

Make separate rule for exceptions like `no-except-else-return`

**Comment by jacobtylerwalls** [comment]
Hey @marcinnformula, thanks for the issue. I'm going to mark this as a duplicate of #8155. Feel free to follow the discussion there.

> avoids negative cons due to possible refactoring in except block when removing return (which would change the program flow).

I agree, and I this applies to if/else also (seems to me an else block is just as likely to have its `return` removed). All this to say, many of the "R" level refactoring messages are really take-them-or-leave-them, and are a matter of taste more than correctness.
